src/effects/VST/VSTEffect.cpp

```diff
@@ -3561,5 +3561,16 @@ bool VSTEffect::TransferDataToWindow(const EffectSettings& settings)
    return true;
 }
 
+EffectSettings VSTEffect::MakeSettings() const
+{
+   auto result = StatefulPerTrackEffect::MakeSettings();
+   // Cause initial population of the map stored in the stateful effect
+   if (!mInitialFetchDone) {
+      FetchSettings(GetSettings(result));
+      mInitialFetchDone = true;
+   }
+   return result;
```

Comment:
This `MakeSettings` is good.  To be very fussy about the correct logical sequence of commits, I would say do this one before defining `TransferDataToWindow`, just so that the lines

```
   if (!StoreSettings(GetSettings(settings)))
      return false;
```

... are sure NOT to return false because `mSettings` has not yet been initialized.

Reply:
Good point, done